docker-compose.yml

```diff
@@ -4,33 +4,98 @@ services:
         build:
             context: .
             dockerfile: ./dockerfiles/Dockerfile
+            target: web-server
         platform: "linux/amd64"
         depends_on:
             - db
             - s3
         ports:
-            - "3000:3000"
-        # for metrics
-        expose: ["3000"]
-        volumes:
-            - "/var/run/docker.sock:/var/run/docker.sock"
-            - ".rustwide-docker:/opt/docsrs/rustwide"
-            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
+            - "3000:80"
         environment:
-            DOCSRS_RUSTWIDE_WORKSPACE: /opt/docsrs/rustwide
+            DOCSRS_PREFIX: /opt/docsrs/prefix
             DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
             DOCSRS_STORAGE_BACKEND: s3
             S3_ENDPOINT: http://s3:9000
             AWS_ACCESS_KEY_ID: cratesfyi
             AWS_SECRET_ACCESS_KEY: secret_key
         env_file:
             - .env
+        healthcheck:
+            test: ["CMD", "curl", "--silent", "--fail", "localhost:80"]
+            interval: 10s
+            timeout: 5s
+            retries: 10
+
+    registry-watcher:
+        build:
+            context: .
+            dockerfile: ./dockerfiles/Dockerfile
+            target: registry-watcher
+        platform: "linux/amd64"
+        depends_on:
+            - db
+        volumes:
+            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
+        environment:
+            DOCSRS_PREFIX: /opt/docsrs/prefix
+            DOCSRS_DATABASE_URL: postgresql://cratesfyi:password@db
+        env_file:
+            - .env
         healthcheck:
             test: ["CMD", "curl", "--silent", "--fail", "localhost:3000"]
             interval: 10s
             timeout: 5s
             retries: 10
 
+    builder:
+        build:
+            context: .
+            dockerfile: ./dockerfiles/Dockerfile
+            target: build-server
+        depends_on:
+            - db
+            - s3
+        volumes:
+            - "/var/run/docker.sock:/var/run/docker.sock"
+            - "cratesio-index:/opt/docsrs/prefix/crates.io-index"
```

Comment:
why does the builder need the index again? What are we using it for? 

`cargo fetch` would use the sparse index protocol, so the only thing I remember is that the builder tries to find the crates.io API URL via the registry, which we could also configure. 

If htere is more we would also have to think about how the builder _updates_ the index, because without the registry watcher there is no update? 

Or am I missing something?

Reply:
Just to look up the API url. I think we should change the builder to use the config (which is already there for the search IIRC) instead of reading it from the index. I'll open a separate PR to do that change before this one.